### PR TITLE
Feature/email jenkins results

### PR DESCRIPTION
**Pull request by chrisveilleux**
## Description
Add steps to Jenkinsfile to email the results of integration tests on success or failure

## How to test
- Make a branch off of this one and create local version of it
- add the branch name to the "when" clause in the integration test stage
- push change to origin

## Contributor license agreement signed?
CLA [yes]

### CODE REVIEW

**Comment by forslund** [comment]
Should we send success-emails? I'd say it's better to only send things when there is something worth noting.

**Comment by forslund** [comment]
Well Github shows the jobs that have finished and their status so I think that information is available and only really of interest to the one reviewing a PR. since a contributor won't have the possibility to access the Jenkins job and restart it afaik.

**Comment by krisgesling** [comment]
I assumed the allure report would be accessible to anyone with the link so they can't restart the job but they could see why their build failed.

**Comment by forslund** [comment]
Yeah exactly, for a failure that's usful but for a success it's redundant imho.

**Comment by forslund** [comment]
I realize now that the results can't be viewed at all except from the link in the e-mail. So maybe it's good to have for now. I do think we should try to get report returned to the the PR instead if the build is successful.

File: Jenkinsfile [hunk 1]
```diff
@@ -73,12 +73,85 @@ pipeline {
...
+                                &nbsp(Requires account on Mycroft's Jenkins instance)
+                            </p>
+                            <br>
+                            <p>
+                                <a href='https://reports.mycroft.ai/core/${BRANCH_ALIAS}'>
+                                    Report of Test Results
+                                </a>
+                            </p>""",
+                        replyTo: 'devops@mycroft.ai',
+                        to: 'devs@mycroft.ai',
```
Comment (forslund) [inline Jenkinsfile:147]: This also sends to devs instead of dev.

File: Jenkinsfile [hunk 2]
```diff
@@ -73,12 +73,85 @@ pipeline {
...
+                        body: """
+                            <p>
+                                All integration tests passed.  No further action required.
+                            </p>
+                            <br>
+                            <p>
+                                <a href='${BUILD_URL}'>
+                                    Jenkins Build Details
+                                </a>
+                                &nbsp(Requires account on Mycroft's Jenkins instance)
```
Comment (forslund) [inline Jenkinsfile:138]: The non-breaking space doesn't render correctly in the e-mail

File: Jenkinsfile [hunk 3]
```diff
@@ -73,12 +73,85 @@ pipeline {
...
+                    // Send success email containing a link to the Jenkins build
+                    // and the results report to Mycroft developers, the developers
+                    // of the pull request and the developers that caused the
+                    // last failed build.
+                    echo 'Sending Success Email'
+                    emailext (
+                        subject: "SUCCESS - Core Integration Tests - Build ${BRANCH_NAME} #${BUILD_NUMBER}",
+                        body: """
+                            <p>
+                                All integration tests passed.  No further action required.
```
Comment (forslund) [inline Jenkinsfile:131]: There's an extra space after the `.`

Comment (chrisveilleux) [reply Jenkinsfile:131]: I am an old, two space after the sentence guy.  I sometimes forget that HTML ignores the second space...

File: Jenkinsfile (hunk 2 above)
Comment (chrisveilleux) [reply Jenkinsfile:138]: Could you be more specific?  I noticed a space in the link after the text but not sure how to get rid of it.

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2530))

**Pull request merged**